benchmarks/GraphBLAS-sharp.Benchmarks/BenchmarksEWiseAdd.fs

```diff
@@ -3,242 +3,628 @@ namespace GraphBLAS.FSharp.Benchmarks
+                            "Configs"
+                            "Context.txt" |]
+            |> Path.GetFullPath
+
+        use reader = new StreamReader(pathToConfig)
+        let platformRegex = Regex <| reader.ReadLine()
+
+        let deviceType =
+            match reader.ReadLine() with
+            | "Cpu" -> DeviceType.Cpu
+            | "Gpu" -> DeviceType.Gpu
+            | "Default" -> DeviceType.Default
+            | _ -> failwith "Unsupported"
+
+        let mutable e = ErrorCode.Unknown
+
+        Cl.GetPlatformIDs &e
+        |> Array.collect (fun platform -> Cl.GetDeviceIDs(platform, deviceType, &e))
+        |> Seq.ofArray
+        |> Seq.distinctBy
+            (fun device ->
+                Cl
+                    .GetDeviceInfo(device, DeviceInfo.Name, &e)
+                    .ToString())
+        |> Seq.filter
+            (fun device ->
+                let platform =
+                    Cl
+                        .GetDeviceInfo(device, DeviceInfo.Platform, &e)
+                        .CastTo<Platform>()
+
+                let platformName =
+                    Cl
+                        .GetPlatformInfo(platform, PlatformInfo.Name, &e)
+                        .ToString()
+
+                platformRegex.IsMatch platformName)
+        |> Seq.map
+            (fun device ->
+                let platform =
+                    Cl
+                        .GetDeviceInfo(device, DeviceInfo.Platform, &e)
+                        .CastTo<Platform>()
+
+                let clPlatform =
+                    Cl
+                        .GetPlatformInfo(platform, PlatformInfo.Name, &e)
+                        .ToString()
+                    |> ClPlatform.Custom
+
+                let deviceType =
+                    Cl
+                        .GetDeviceInfo(device, DeviceInfo.Type, &e)
+                        .CastTo<DeviceType>()
+
+                let clDeviceType =
+                    match deviceType with
+                    | DeviceType.Cpu -> ClDeviceType.CPU
+                    | DeviceType.Gpu -> ClDeviceType.GPU
+                    | DeviceType.Default -> ClDeviceType.Default
+                    | _ -> failwith "Unsupported"
+
+                ClContext(clPlatform, clDeviceType))
+
+type EWiseAddBenchmarks4Float32COO() =
+    inherit EWiseAddBenchmarks()
+
+    let mutable leftCOO =
+        Unchecked.defaultof<Backend.COOMatrix<float32>>
+
+    let mutable rightCOO =
+        Unchecked.defaultof<Backend.COOMatrix<float32>>
+
+    let mutable resultCOO =
+        Unchecked.defaultof<Backend.COOMatrix<float32>>
+
+    member val FirstMatrix = Unchecked.defaultof<COOMatrix<float32>> with get, set
+    member val SecondMatrix = Unchecked.defaultof<COOMatrix<float32>> with get, set
+
+    member this.Processor = this.OclContext.Provider.CommandQueue
+    // To do: set the correct workGroupSize
+    member this.eWiseAdd =
+        Backend.COOMatrix.eWiseAdd (this.OclContext) <@ (+) @> 128 this.Processor
```

Comment:
Looks like it should be local ```let``` binding. It allows compiling a kernel on a class constructor call.